src/API.js

```diff
@@ -57,11 +79,63 @@ async function _fetch(path, options = {}) {
         method: 'GET',
         ...options
     })
-    
+
     if (!response.ok) {
         const error = (data && data.message) || response.status;
         throw new Error(error);
     }
 
     return response;
-};
\ No newline at end of file
+};
+
+function formatURL(url, replacements) {
+    let str = url.replace(/%\w+%/g, function (all) {
+        return replacements[all] || all;
+    });
+
+    return str;
+}
+
+function getFormattedURL(replacements, endpoint) {
+    switch (endpoint) {
+        case endpoints.regInit:
+            return formatURL(baseUrl + endpoints.regInit, replacements);
+        case endpoints.regSubmit:
+            return formatURL(baseUrl + endpoints.regSubmit, replacements);
+        case endpoints.loginInit:
+            return formatURL(baseUrl + endpoints.loginInit, replacements);
+        case endpoints.loginSubmit:
+            return formatURL(baseUrl + endpoints.loginSubmit, replacements);
+        case endpoints.logoutInit:
+            return formatURL(baseUrl + endpoints.logoutInit, replacements);
+        case endpoints.logoutSubmit:
+            return formatURL(baseUrl + endpoints.logoutSubmit, replacements);
+        case endpoints.whoami:
+            return formatURL(baseUrl + endpoints.whoami, replacements);
+        default:
+            return "";
+    }
+}
+
+function getFormattedProxyURL(replacements, endpoint) {
+    switch (endpoint) {
+        case endpoints.regInit:
+            return formatURL(proxyUrl + endpoints.regInit, replacements);
+        case endpoints.regSubmit:
+            return formatURL(proxyUrl + endpoints.regSubmit, replacements);
+        case endpoints.loginInit:
+            return formatURL(proxyUrl + endpoints.loginInit, replacements);
+        case endpoints.loginSubmit:
+            return formatURL(proxyUrl + endpoints.loginSubmit, replacements);
+        case endpoints.logoutInit:
+            return formatURL(proxyUrl + endpoints.logoutInit, replacements);
+        case endpoints.logoutSubmit:
+            return formatURL(proxyUrl + endpoints.logoutSubmit, replacements);
+        case endpoints.whoami:
+            return formatURL(proxyUrl + endpoints.whoami, replacements);
+        default:
+            return "";
+    }
+}
+
+export { getFormattedURL, getFormattedProxyURL };
```

Comment:
```suggestion
export function getFormattedURL(replacements, endpoint, base = baseUrl) {
    if (!endpoints.hasOwnProperty(endpoint)) {
        return "";
    }

    return formatURL(base + endpoints[endpoint], replacements);
}

export function getFormattedProxyURL(replacements, endpoint) {
    return getFormattedURL(replacements, endpoint, proxyUrl);
}

```

Make this more generic, so the `endpoints` object is our single source of truth.